Source/WebKit/UIProcess/API/C/curl/WKWebsiteDataStoreConfigurationRefCurl.cpp

```diff
@@ -0,0 +1,40 @@
+/*
+ * Copyright (C) 2023 Sony Interactive Entertainment Inc.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions
+ * are met:
+ * 1. Redistributions of source code must retain the above copyright
+ *    notice, this list of conditions and the following disclaimer.
+ * 2. Redistributions in binary form must reproduce the above copyright
+ *    notice, this list of conditions and the following disclaimer in the
+ *    documentation and/or other materials provided with the distribution.
+ *
+ * THIS SOFTWARE IS PROVIDED BY APPLE INC. AND ITS CONTRIBUTORS ``AS IS''
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO,
+ * THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR
+ * PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL APPLE INC. OR ITS CONTRIBUTORS
+ * BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+ * CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ * SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF
+ * THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#include "config.h"
+#include "WKWebsiteDataStoreConfigurationRefCurl.h"
+
+#include "WKAPICast.h"
+#include "WebsiteDataStoreConfiguration.h"
+
+WKStringRef WKWebsiteDataStoreConfigurationCopyAlternativeServicesStorageFile(WKWebsiteDataStoreConfigurationRef configuration)
```

Comment:
I will check it.

Reply:
It looks like the dataStoreForIdentifier function creates a UUID directory under NSLibrarydirectory. However, on other platforms, users may want to change the directory location, which corresponds to the NSLibraryDirectory.

https://github.com/WebKit/WebKit/blob/0646de88e5b9e9e657741c81b75e51e353ff35ae/Source/WebKit/UIProcess/WebsiteData/WebsiteDataStore.cpp#L262

https://github.com/WebKit/WebKit/blob/0646de88e5b9e9e657741c81b75e51e353ff35ae/Source/WebKit/UIProcess/WebsiteData/WebsiteDataStoreConfiguration.cpp#L62-L63

https://github.com/WebKit/WebKit/blob/0646de88e5b9e9e657741c81b75e51e353ff35ae/Source/WebKit/UIProcess/WebsiteData/Cocoa/WebsiteDataStoreCocoa.mm#L312-L315

https://github.com/WebKit/WebKit/blob/0646de88e5b9e9e657741c81b75e51e353ff35ae/Source/WebKit/UIProcess/WebsiteData/Cocoa/WebsiteDataStoreCocoa.mm#L252

Reply:
Would it be okay to add the WKWebsiteDataStoreConfigurationCreateWithBaseDirectory function to WKWebsiteDataStoreConfiurationRef, as it would be convenient to be able to specify the base directory on the other ports?

https://github.com/WebKit/WebKit/blob/0646de88e5b9e9e657741c81b75e51e353ff35ae/Source/WebKit/UIProcess/WebsiteData/WebsiteDataStoreConfiguration.h#L55-L59

https://github.com/WebKit/WebKit/blob/0646de88e5b9e9e657741c81b75e51e353ff35ae/Source/WebKit/UIProcess/WebsiteData/WebsiteDataStoreConfiguration.cpp#L80-L81

https://github.com/WebKit/WebKit/blob/0646de88e5b9e9e657741c81b75e51e353ff35ae/Source/WebKit/UIProcess/API/C/WKWebsiteDataStoreConfigurationRef.h#L37

Reply:
Then, we save the altsvc-chache.txt file under the m_baseCacheDirectory directory.